### PR TITLE
feat(pyroscope/scrape): add support for configuring CPU profile's duration scraped by `pyroscope.scrape`

### LINKED ISSUE #195: Add a configuration option to pyroscope.scrape to define for how long a CPU profile should run
### Request

Right now, the only configuration option to customize scraping interval for CPU collection are `scrape_interval` and `delta`. If we want to use delta profiling, but reduce the amount of profiles we collect, there's no other options.

As the doc says 
> The seconds used will be equal to scrape_interval - 1. For example, if scrape_interval is "15s", seconds will be 14 seconds. If the HTTP endpoint is /debug/pprof/profile, then the HTTP query will become /debug/pprof/profile?seconds=14

- If we set a larger scrape interval (60s), `seconds` will be 59
- If we set a smaller scrape interval (15s), `seconds` will be 14

With this param, we are basically running profiles all the time (except for this 1s between profiles)

We could add a configuration option to let the user decide of this `seconds` value. 

for instance
```
scrape_interval = 15s
scrape_duration = 5s
```

would result in a call to `/debug/pprof/profile?seconds=5` every 15s. This would enable considerable savings on resources used by grafana-agent and pyroscope, if it's a tradeoff the user is willing to accept

### Use case

User might want to reduce the amount of CPU profiles collected in order to save on resources (CPU, network, storage...)

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Fixes #195 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by clayton-cornell** [comment]
Docs here include suggestions from the Agent PR, so all good from a doc standpoint. Over to the @grafana/grafana-alloy-maintainers  team for code review

File: docs/sources/reference/components/pyroscope.scrape.md [hunk 1]
```diff
@@ -395,8 +396,10 @@ When the `delta` argument is `false`, the [pprof][] HTTP query will be instantan
 When the `delta` argument is `true`:
 * The [pprof][] HTTP query will run for a certain amount of time.
 * A `seconds` parameter is automatically added to the HTTP request.
-* The `seconds` used will be equal to `scrape_interval - 1`.
-  For example, if `scrape_interval` is `"15s"`, `seconds` will be 14 seconds.
+* The default value for the `seconds` query parameter is `scrape_interval - 1`.
+    If you set `profiling_duration`, then `seconds` is assigned the same value as `profiling_duration`.
+    For example, if you set `scrape_interval` to `"15s"`, then `seconds` defaults to `14s`.
+    If you set `profiling_duration` to `16s`, then `seconds is set to `16s` regardless of the `scrape_interval` value.
```
Comment (tpaschalis) [inline docs/sources/reference/components/pyroscope.scrape.md:402]: Just a missing backtick.

```suggestion
    If you set `profiling_duration` to `16s`, then `seconds` is set to `16s` regardless of the `scrape_interval` value.
```

File: docs/sources/reference/components/pyroscope.scrape.md [hunk 2]
```diff
@@ -395,8 +396,10 @@ When the `delta` argument is `false`, the [pprof][] HTTP query will be instantan
 When the `delta` argument is `true`:
 * The [pprof][] HTTP query will run for a certain amount of time.
 * A `seconds` parameter is automatically added to the HTTP request.
-* The `seconds` used will be equal to `scrape_interval - 1`.
-  For example, if `scrape_interval` is `"15s"`, `seconds` will be 14 seconds.
+* The default value for the `seconds` query parameter is `scrape_interval - 1`.
+    If you set `profiling_duration`, then `seconds` is assigned the same value as `profiling_duration`.
+    For example, if you set `scrape_interval` to `"15s"`, then `seconds` defaults to `14s`.
+    If you set `profiling_duration` to `16s`, then `seconds` is set to `16s` regardless of the `scrape_interval` value.
```
Comment (tpaschalis) [inline docs/sources/reference/components/pyroscope.scrape.md:402]: cc @simonswine @korniltsev does it make sense for the `profiling_duration` to ever be larger than the `scrape_interval` here? Or similar to the Prometheus scrape_timeout it should be always less than the scrape interval?

File: internal/component/pyroscope/scrape/scrape.go [hunk 3]
```diff
@@ -21,15 +21,17 @@ import (
...
+	pprofBlock               string        = "block"
+	pprofGoroutine           string        = "goroutine"
+	pprofMutex               string        = "mutex"
+	pprofProcessCPU          string        = "process_cpu"
+	pprofFgprof              string        = "fgprof"
+	pprofGoDeltaProfMemory   string        = "godeltaprof_memory"
+	pprofGoDeltaProfBlock    string        = "godeltaprof_block"
+	pprofGoDeltaProfMutex    string        = "godeltaprof_mutex"
+	defaultScrapeInterval    time.Duration = 15 * time.Second
+	defaultProfilingDuration time.Duration = 14 * time.Second
```
Comment (tpaschalis) [inline internal/component/pyroscope/scrape/scrape.go:34]: We're not using defaultScrapeInterval anywhere.

If we want to provide a sensible default, should we make it like this and wire the defaultScrapeInterval in NewDefaultArguments?

```suggestion
	defaultScrapeInterval    time.Duration = 15 * time.Second
	defaultProfilingDuration time.Duration = defaultScrapeInterval - 1 * time.Second
```

**Review by tpaschalis — COMMENTED** [review]
This is on a good path. I'd like some opinion from the Pyroscope folks on whether we can leave the `profiling_value` set to any as this would mean that we could effectively be _always_ profiling multiple times in parallel which doesn't make much sense to me.

File: docs/sources/reference/components/pyroscope.scrape.md (hunk 2 above)
Comment (julienvey) [reply docs/sources/reference/components/pyroscope.scrape.md:402]: It should always be less. If you try to profile a service which is already running a profile, you'll get an error
```
server returned HTTP status (500) Could not enable CPU profiling: cpu profiling already in use
```
I think we should either add some validation, or default back to `min(scrape_interval - 1, profiling_duration)`

Comment (tpaschalis) [reply docs/sources/reference/components/pyroscope.scrape.md:402]: I think that makes sense as well, yeah cc @hainenber

Comment (hainenber) [reply docs/sources/reference/components/pyroscope.scrape.md]: I've added the validation in the HEAD. Looks A-OK to you guys?

File: docs/sources/reference/components/pyroscope.scrape.md [hunk 4]
```diff
@@ -65,6 +65,7 @@ Name                | Type                     | Description
 `params`            | `map(list(string))`      | A set of query parameters with which the target is scraped.        |                | no
 `scrape_interval`   | `duration`               | How frequently to scrape the targets of this scrape configuration. | `"15s"`        | no
 `scrape_timeout`    | `duration`               | The timeout for scraping targets of this configuration. Must be larger than `scrape_interval`. | `"18s"` | no
+`profiling_duration`| `duration` | The duration for a delta profiling to be scraped. Must be larger than 1 second. | `"14s"` | no
```
Comment (tpaschalis) [inline docs/sources/reference/components/pyroscope.scrape.md:68]: Sorry, one last thing. Could we name this with a little more descriptive name?

```suggestion
`delta_profiling_duration`| `duration` | The duration for a delta profiling to be scraped. Must be larger than 1 second. | `"14s"` | no
```

**Comment by simonswine** [comment]
Thanks for the contribution, this is a very useful knob to be able to turn 🙂 

I am happy with this to go forward, once all the points @tpaschalis has raised are addressed.

**Comment by hainenber** [comment]
Oops, `git rebase` sheenanigan 🤦

File: docs/sources/reference/components/pyroscope.scrape.md (hunk 4 above)
Comment (hainenber) [reply docs/sources/reference/components/pyroscope.scrape.md]: I've changed this in HEAD. PTAL

**Comment by julienvey** [comment]
@tpaschalis @simonswine Anything we can do to help move this forward ? I think @hainenber fixed all the previous comment

**Comment by salarali** [comment]
Has this change been released? I don't see it in the docs or the changelogs.

**Comment by clayton-cornell** [comment]
> Has this change been released? I don't see it in the docs or the changelogs.

It is in the `next` release, so for docs, you find the new content in the "next" version of the `pyroscope.scrape` component reference documentation: https://grafana.com/docs/alloy/next/reference/components/pyroscope.scrape/#common-configuration  When Alloy v1.2 is release, this doc change will show up in "latest".

**Pull request merged**